csharp/extractor/Semmle.Extraction.CSharp/Extractor/Extractor.cs

```diff
@@ -129,6 +134,42 @@ public static ExitCode Run(string[] args)
             }
         }
 
+        private static void AddSourceFilesFromProjects(IEnumerable<string> projectsToLoad, IList<string> compilerArguments, ILogger logger)
+        {
+            logger.Log(Severity.Info, "  Loading referenced projects.");
+            var projects = new Queue<string>(projectsToLoad);
+            var processed = new HashSet<string>();
+            while (projects.Count > 0)
+            {
+                var project = projects.Dequeue();
+                var fi = new FileInfo(project);
+                if (processed.Contains(fi.FullName))
+                {
+                    continue;
+                }
+
+                processed.Add(fi.FullName);
+                logger.Log(Severity.Info, "  Processing referenced project: " + fi.FullName);
+
+                var csProj = new CsProjFile(fi);
+
+                foreach (var cs in csProj.Sources)
+                {
+                    if (cs.EndsWith("obj/Debug/.NETCoreApp,Version=v5.0.AssemblyAttributes.cs") ||
```

Comment:
The generated .csproj files are [now hard coded](https://github.com/github/codeql/pull/5664/files?file-filters%5B%5D=.py#diff-37daa16469133530ed373f2fce5f58d735f83fd4ff8d3dcdc5cab49a66be6113R132) to `<TargetFramework>net5.0</TargetFramework>`, so this filtering will work. But it would be great if we didn't have to update all stub .csproj files and this bit of code when we move to newer version of .net. 

I think the file name is generated by [this code](https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L3417).

Actually, what do you think, shouldn't we simply exclude all files from the `obj` folders?